lib/fog/ovirt/requests/compute/v4/update_vm.rb

```diff
@@ -0,0 +1,56 @@
+module Fog
+  module Compute
+    class Ovirt
+      class V4
+        class Real
+          # rubocop:disable Metrics/AbcSize, Metrics/PerceivedComplexity
+          def update_vm(attrs)
+            attrs = attrs.dup
+            vm_service = client.system_service.vms_service.vm_service(attrs[:id])
+
+            if attrs[:cores].present?
+              cpu_topology = OvirtSDK4::CpuTopology.new(:cores => attrs[:cores], :sockets => "1")
+              attrs[:cpu] = OvirtSDK4::Cpu.new(:topology => cpu_topology)
+            end
+            wrap_attribute(attrs, :cluster, OvirtSDK4::Cluster)
+            attrs[:template] = if attrs[:template].present?
+                                 OvirtSDK4::Template.new(:id => attrs[:template])
+                               else
+                                 blank_template
+                               end
+
+            wrap_attribute(attrs, :large_icon, OvirtSDK4::Icon)
+            wrap_attribute(attrs, :small_icon, OvirtSDK4::Icon)
+            wrap_attribute(attrs, :cpu_profile, OvirtSDK4::CpuProfile)
+            wrap_attribute(attrs, :host, OvirtSDK4::Host)
+            wrap_attribute(attrs, :quota, OvirtSDK4::Quota)
+            attrs[:high_availability] = OvirtSDK4::HighAvailability.new(:enabled => attrs[:ha]) unless attrs[:ha].nil?
+            attrs[:instance_type] = attrs[:instance_type].present? ? OvirtSDK4::InstanceType.new(:id => attrs[:instance_type]) : nil
+            attrs[:original_template] = if attrs[:original_template].present?
+                                          OvirtSDK4::Template.new(:id => attrs[:original_template])
+                                        else
+                                          blank_template
+                                        end
+            vm_service.update(attrs)
+          end
+          # rubocop:enable Metrics/AbcSize, Metrics/PerceivedComplexity
+
+          def wrap_attribute(attrs, klass, attribute)
+            if attrs[attribute].present?
+              attrs[attribute] = klass.new(:id => attrs[attribute])
+            else
+              attrs.delete(input_attribute)
```

Comment:
This fails with:

```
Failed to power up a compute abcde ovirt (oVirt) instance ray-yokoyama.inecas.ovirt.test: undefined local variable or method `input_attribute' for #<Fog::Compute::Ovirt::V4::Real:0x0055793befd5f8>
 | NameError: undefined local variable or method `input_attribute' for #<Fog::Compute::Ovirt::V4::Real:0x0055793befd5f8>
 | /home/inecas/Projects/ws/foreman-rex/fog-ovirt/lib/fog/ovirt/requests/compute/v4/update_vm.rb:42:in `wrap_attribute'
 | /home/inecas/Projects/ws/foreman-rex/fog-ovirt/lib/fog/ovirt/requests/compute/v4/update_vm.rb:15:in `update_vm'
 | /home/inecas/Projects/ws/foreman-rex/fog-ovirt/lib/fog/ovirt/models/compute/server.rb:177:in `save'
 | /home/inecas/Projects/ws/foreman-rex/foreman-rails5/app/models/compute_resources/foreman/model/ovirt.rb:226:in `start_vm'
 | /home/inecas/Projects/ws/foreman-rex/foreman-rails5/app/models/concerns/orchestration/compute.rb:201:in `setComputePowerUp'
 | /home/inecas/Projects/ws/foreman-rex/foreman-rails5/app/models/concerns/orchestration.rb:222:in `execute'
 | /home/inecas/Projects/ws/foreman-rex/foreman-rails5/app/models/concerns/orchestration.rb:145:in `block in process'
```

Reply:
leftover :\ sorry about that, fixing now.